test/test_ops_gradients.py

```diff
@@ -141,6 +148,22 @@ def test_fn_gradgrad(self, device, dtype, op):
             self.skipTest("Skipped! Operation does not support gradgrad")
         self._check_helper(device, dtype, op, op.get_op(), 'bwgrad_bwgrad')
 
+    @skipCUDAIfRocm
+    @_gradcheck_ops(_python_ref_ops)
+    def test_fn_gradgrad_python_references(self, device, dtype, op):
+        self._skip_helper(op, device, dtype)
+        if not op.supports_gradgrad:
+            self.skipTest("Skipped! Operation does not support gradgrad")
+        self._check_helper(device, dtype, op, op.python_ref, 'bwgrad_bwgrad')
+
+    # Test that gradients of gradients are computed correctly
+    @_gradcheck_ops(op_db)
+    def test_fn_gradgrad(self, device, dtype, op):
+        self._skip_helper(op, device, dtype)
+        if not op.supports_gradgrad:
+            self.skipTest("Skipped! Operation does not support gradgrad")
+        self._check_helper(device, dtype, op, op.get_op(), 'bwgrad_bwgrad')
```

Comment:
@mruberry why are there separate tests for the references? I was expecting something like the following:

1. each reference implementation gets its own OpInfo. The name would be "torch.refs.add", for example.
2. The metadata for this OpInfo (sample_inputs, etc) somehow gets autogenerated from the torch.add OpInfo.
3. Since torch.refs.add is an OpInfo, it'll run through ALL existing opinfo-based tests.


This approach, "write an OpInfo per reference", would make it so that we test the reference implementation on more than just the tests in this PR. I'm thinking about the CompositeCompliance tests (and also functorch tests, if functorch were to use the reference implementations)

Reply:
Having a new OpInfo that inherits from the previous one sounds like a good idea. We need to be able to selectively override the referenced operation's metadata, too, and that would be an elegant way to support it.